server/src/main/java/io/crate/metadata/upgrade/MetadataIndexUpgrader.java

```diff
@@ -109,4 +115,20 @@ private void upgradeColumnPositions(Map<String, Object> defaultMap, @Nullable In
             IndexTemplateUpgrader.populateColumnPositions(defaultMap);
         }
     }
+
+    // bug : https://github.com/crate/crate/issues/13380
+    private String fixIndexName(String indexName) {
+        if (indexName.startsWith(".partitioned")) {
+            try {
+                new IndexParts(indexName);
+            } catch (IllegalArgumentException e) {
+                String[] indexParts = indexName.split("\\.");
+                indexParts[0] = indexParts[2];
+                indexParts[1] = ".partitioned";
+                indexParts[2] = null;
```

Comment:
Is this really correct as it looks like it will end up in a partitioned table with a schema set but no table name? 
Could we create unit tests for this method with variants which would also clarify the expected outcome?